api/messages.proto

```diff
@@ -202,7 +202,9 @@ message ApplicationUplink {
 
   bool confirmed = 11;
 
-  // next: 13
+  // End device location metadata, set by the Application Server while handling the message.
+  map<string,Location> locations = 13;
```

Comment:
Why the `map` instead of `repeated` ? `Location` contains the source.

Reply:
This is the same format as in `EndDevice` https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.10/api/end_device.proto#L437.